src/main/java/com/icthh/xm/tmf/ms/communication/service/mail/EmailTemplateService.java

```diff
@@ -102,11 +104,20 @@ public void updateTemplate(String templateKey, String langKey, UpdateTemplateReq
     }
 
     private TemplateDetails createTemplateDetails(String templateContent, EmailTemplateSpec emailTemplateSpec,
-                                                  List<String> langs, String subjectTemplate) {
+                                                  List<String> langs, String subjectTemplate, String langKey) {
         TemplateDetails templateDetails = templateDetailsMapper.emailTemplateToDetails(emailTemplateSpec);
         templateDetails.setContent(templateContent);
         templateDetails.setSubjectTemplate(subjectTemplate);
         templateDetails.setLangs(langs);
+
+        List<String> dependsOnTemplateKeys = emailTemplateSpec.getDependsOnTemplateKeys();
```

Comment:
add check for case when dependsOnTemplateKeys is null

Reply:
added